### PR TITLE
Sorting

### PR DESCRIPTION
**Pull request by shahidulkareem**
The **`columns`** prop is an array of column objects which have a key ,label and **`sort`**. Like this:

```javascript
const columns = [{
    key: 'userId',
    label: '#',
    sort: false
  }, {
    key: 'name',
    label: 'Name',
   sort: true
  }, {
    key: 'email',
    label: 'Email',
    sort: false
  },
];
```

### CODE REVIEW

**Review by shahidulkareem — COMMENTED** [review]
I have implemented sorting functionality.

File: src/spicy-datatable/SpicyDatatable.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import DatatableHeader from './components/DatatableHeader.js';
 import DatatableRows from './components/DatatableRows.js';
 import Pagination from './components/Pagination.js';
 import { SpicyDatatablePropTypes } from './PropTypes.js';
-import { filterRows, getSafely, setSafely } from './helpers';
+import { sort_by, filterRows, getSafely, setSafely } from './helpers';
```
Comment (filipdanic) [inline src/spicy-datatable/SpicyDatatable.js]: Can you please update this to `sortBy` to follow the naming convention?

File: src/spicy-datatable/helpers.js [hunk 2]
```diff
@@ -80,3 +80,19 @@ export function filterRows(rows, columns, searchQuery = '') {
...
+ * @param {String} field
+ * @param {boolean} reverse
+ * @param {return function} primer
+ */
+export function sort_by(field, reverse, primer) {
+   var key = function (x) {return primer ? primer(x[field]) : x[field]};
+
+   return function (a,b) {
+	  var A = key(a), B = key(b);
+	  return ( (A < B) ? -1 : ((A > B) ? 1 : 0) ) * [-1,1][+!!reverse];
```
Comment (filipdanic) [inline src/spicy-datatable/helpers.js]: This line is hard to follow. Can you refactor this function to make it easier to understand and maintain?

File: src/spicy-datatable/helpers.js [hunk 3]
```diff
@@ -80,3 +80,19 @@ export function filterRows(rows, columns, searchQuery = '') {
...
+
+
+/**
+ * Sort the rows for the column.
+ * @param {String} field
+ * @param {boolean} reverse
+ * @param {return function} primer
+ */
+export function sort_by(field, reverse, primer) {
+   var key = function (x) {return primer ? primer(x[field]) : x[field]};
```
Comment (filipdanic) [inline src/spicy-datatable/helpers.js]: Can we use arrow functions/const like the rest of the library? E.g: `const key = x => primer ? primer(x[filed]) : x[field]`

File: src/spicy-datatable/SpicyDatatable.js [hunk 4]
```diff
@@ -112,6 +114,25 @@ class SpicyDatatable extends Component {
...
+  
+  handleSort = (e, c) => {
+  	const { rows } = this.props;
+    const column = c.key;
+    const { tableKey } = this.props;
+    const { filteredRows, sortColumn, sortOrder } = this.state;
+    if (this.scheduleQueryChange) {
+      clearTimeout(this.scheduleQueryChange);
+    }
+    this.scheduleQueryChange = setTimeout(() => {
```
Comment (filipdanic) [inline src/spicy-datatable/SpicyDatatable.js:126]: Good call on using debouncing here, but the function inside of the `setTimeout` is hard to read and follow. Any way you could refactor this? E.g. extract the conditionals as a separate function call and using arrow syntax for the function call on line 129.

**Review by filipdanic — CHANGES_REQUESTED** [review]

**Comment by shahidulkareem** [comment]
Thanks, I have resolved all your requested changes.Please have a look.

File: src/spicy-datatable/SpicyDatatable.js (hunk 1 above)
Comment (shahidulkareem) [reply src/spicy-datatable/SpicyDatatable.js]: changed <code>sort_by</code> as  <code>sortBy </code>

File: src/spicy-datatable/helpers.js (hunk 2 above)
Comment (shahidulkareem) [reply src/spicy-datatable/helpers.js]: Now i have changed this line as separate const function. Like below,
<code>const result = (row1,row2) => ( (row1 < row2) ? -1 : ((row1 > row2) ? 1 : 0) ) * [-1,1][Number(Boolean(sortOrder))];</code>

File: src/spicy-datatable/helpers.js (hunk 3 above)
Comment (shahidulkareem) [reply src/spicy-datatable/helpers.js]: Changed arrow functions as cons in rest of the library. Like Below
<code>const lowerCase = (value) => typeof value === "string" ? value.toLowerCase() : value;
	const key = (row) => lowerCase ? lowerCase(row[sortColumn]) : row[sortColumn]; </code>

File: src/spicy-datatable/SpicyDatatable.js (hunk 4 above)
Comment (shahidulkareem) [reply src/spicy-datatable/SpicyDatatable.js:126]: Changed separate function call. Like below,

<code> const sortFunction = () => sortRows.sort(sortBy(column, sortRev));
      const filteredSortRows = (column.length === 0 ? [] : sortFunction()) || [];</code>
